jmxtrans-core/src/main/java/com/googlecode/jmxtrans/cli/JmxTransConfiguration.java

```diff
@@ -144,10 +169,138 @@ public File apply(String input) {
+			},
+			new PropertySetter<File>(CONFIG_FILE_PROPERTY, File.class) {
+				@Override
+				protected void doSetValue(File value, JmxTransConfiguration configuration) {
+					configuration.setConfigFile(value);
+				}
+			},
+			new PropertySetter<Boolean>(RUN_ENDLESSLY_PROPERTY, Boolean.class) {
+				@Override
+				protected void doSetValue(Boolean value, JmxTransConfiguration configuration) {
+					configuration.setRunEndlessly(value);
+				}
+			},
+			new PropertySetter<File>(QUARTZ_PROPERTIES_FILE_PROPERTY, File.class) {
+				@Override
+				protected void doSetValue(File value, JmxTransConfiguration configuration) {
+					configuration.setQuartzPropertiesFile(value);
+				}
+			},
+			new PropertySetter<Integer>(RUN_ENDLESSLY_PROPERTY, Integer.class) {
+				@Override
+				protected void doSetValue(Integer value, JmxTransConfiguration configuration) {
+					configuration.setRunPeriod(value);
+				}
+			},
+			new PropertySetter<String>(ADDITIONAL_JARS_PROPERTY, String.class) {
+				@Override
+				protected void doSetValue(String value, JmxTransConfiguration configuration) {
+				}
+			},
+			new PropertySetter<Integer>(QUERY_PROCESSOR_EXECUTOR_POOL_SIZE_PROPERTY, Integer.class) {
+				@Override
+				protected void doSetValue(Integer value, JmxTransConfiguration configuration) {
+					configuration.setQueryProcessorExecutorPoolSize(value);
+				}
+			},
+			new PropertySetter<Integer>(QUERY_PROCESSOR_EXECUTOR_WORK_QUEUE_CAPACITY_PROPERTY, Integer.class) {
+				@Override
+				protected void doSetValue(Integer value, JmxTransConfiguration configuration) {
+					configuration.setQueryProcessorExecutorWorkQueueCapacity(value);
+				}
+			},
+			new PropertySetter<Integer>(RESULT_PROCESSOR_EXECUTOR_POOL_SIZE_PROPERTY, Integer.class) {
+				@Override
+				protected void doSetValue(Integer value, JmxTransConfiguration configuration) {
+					configuration.setResultProcessorExecutorPoolSize(value);
+				}
+			},
+			new PropertySetter<Integer>(RESULT_PROCESSOR_EXECUTOR_WORK_QUEUE_CAPACITY_PROPERTY, Integer.class) {
+				@Override
+				protected void doSetValue(Integer value, JmxTransConfiguration configuration) {
+					configuration.setResultProcessorExecutorWorkQueueCapacity(value);
+				}
+			},
+			new PropertySetter<Boolean>(USE_SEPARATE_EXECUTORS_PROPERTY, Boolean.class) {
+				@Override
+				protected void doSetValue(Boolean value, JmxTransConfiguration configuration) {
+					configuration.setUseSeparateExecutors(value);
+				}
+			}
+	};
+
+
+	@VisibleForTesting
+	void loadProperties(Properties properties) {
+		TypedProperties typedProperties = new TypedProperties(properties);
+		for(PropertySetter setter: SETTERS) {
+			setter.setValue(typedProperties, this);
+		}
+	}
+
+	public void loadProperties(File configFile) throws IOException {
+		Properties properties = new Properties();
+		try (InputStream in = this.getClass().getClassLoader().getResourceAsStream("jmxtrans.properties")) {
+			properties.load(in);
+		}
+		if (configFile != null) {
```

Comment:
We could probably reduce duplication and increase readability by extracting a method to find the right file to load and one to load it.

Reply:
This does property merging, you can override just some properties.
The order of priority is now (from most important to less important):

1. Command line args
2. Provided properties file or if not provided and exists default jmxtrans.properties
3. jmxtrans.properties in classpath
4. Default values in JmxtransConfiguration